core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,79 @@ def test_user_contributions_get_created_after_running_the_job(self):
             ['exp_id'])
 
 
+class UserAuthModelOneOffJobTests(test_utils.GenericTestBase):
+    """Tests for the one-off UserAuthModel migration job."""
+
+    USER_A_EMAIL = 'a@example.com'
+    USER_A_ID = 'user_a_id'
+    USER_A_GAE_ID = 'user_a_gae_id'
+    USER_B_EMAIL = 'b@example.com'
+    USER_B_ID = 'user_b_id'
+    USER_B_GAE_ID = 'user_b_gae_id'
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = user_jobs_one_off.UserAuthModelOneOffJob.create_new()
+        user_jobs_one_off.UserAuthModelOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+        output = (
+            user_jobs_one_off.UserAuthModelOneOffJob.get_output(
+                job_id))
+        return output
+
+    def setUp(self):
+        super(UserAuthModelOneOffJobTests, self).setUp()
+        self.user_a_model = user_models.UserSettingsModel(
+            id=self.USER_A_ID,
+            gae_id=self.USER_A_GAE_ID,
+            email=self.USER_A_EMAIL,
+        )
+        self.user_b_model = user_models.UserSettingsModel(
+            id=self.USER_B_ID,
+            gae_id=self.USER_B_GAE_ID,
+            email=self.USER_B_EMAIL
+        )
+        self.user_a_model.put()
+
+    def test_before_migration_user_does_not_exist_in_user_auth_model(self):
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_B_ID))
+
+    def test_one_off_job_migrates_users_without_any_error(self):
+        self.user_b_model.put()
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_B_ID))
+
+        self._run_one_off_job()
+
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_A_GAE_ID)
+        self.assertEqual(user_auth_model.id, self.USER_A_ID)
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_B_GAE_ID)
+        self.assertEqual(user_auth_model.id, self.USER_B_ID)
+
+    def test_one_off_job_gae_id_registered_to_multiple_users_raises_error(self):
+        self._run_one_off_job()
+        self.user_b_model.gae_id = self.USER_A_GAE_ID
+        self.user_b_model.put()
+        output = self._run_one_off_job()
+        self.assertEqual(len(output), 1)
+        expected_error = python_utils.UNICODE(
+            'GAE_ID %s already registered with user_id %s.' % (
+                self.USER_A_GAE_ID, self.user_a_model.id))
+        self.assertEqual(expected_error, output[0])
+
+    def test_one_off_job_run_multiple_times_raises_no_error(self):
```

Comment:
Shouldn't you be checking that there are no expected errors from the output of the jobs? It seems like this test will always pass as-is.

Reply:
Did that check now. PTAL!

Reply:
Looks good--thanks!